### PR TITLE
Improve performance by removing unused assignments

### PR DESCRIPTION
**Pull request by rbuj**

### CODE REVIEW

File: src/core/boxes.c [hunk 1]
```diff
@@ -577,7 +577,6 @@ meta_rectangle_get_minimal_spanning_set_for_region (
   *temp_rect = *basic_rect;
   ret = g_list_prepend (NULL, temp_rect);
 
-  strut_iter = all_struts;
```
Comment (muktupavels) [inline src/core/boxes.c]: This most likely is optimized out by compiler... So I doubt it improves anything.

File: src/core/edge-resistance.c [hunk 2]
```diff
@@ -285,7 +285,6 @@ find_nearest_position (const GArray        *edges,
           if (dist < best_dist)
             {
               best = compare;
-              best_dist = dist;
```
Comment (muktupavels) [inline src/core/edge-resistance.c]: Care to explain how this is unused?

File: src/core/window.c [hunk 3]
```diff
@@ -3597,11 +3597,6 @@ meta_window_move_resize_internal (MetaWindow          *window,
                   window->frame->rect.width,
                   window->frame->rect.height);
     }
-  else
-    {
-      frame_size_dx = 0;
```
Comment (muktupavels) [inline src/core/window.c]: How about reducing scope by moving `frame_size_dx` and `frame_size_dy` into if branch if they are used only there?

File: src/ui/frames.c [hunk 4]
```diff
@@ -1492,8 +1492,6 @@ meta_frames_button_press_event (GtkWidget      *widget,
           op = op > META_GRAB_OP_CLICKING_MAXIMIZE_HORIZONTAL ? META_GRAB_OP_CLICKING_MAXIMIZE : op;
           break;
         case META_FRAME_CONTROL_UNMAXIMIZE:
-          op = META_GRAB_OP_CLICKING_UNMAXIMIZE + event->button - 1;
-          op = op > META_GRAB_OP_CLICKING_UNMAXIMIZE_HORIZONTAL ? META_GRAB_OP_CLICKING_UNMAXIMIZE : op;
```
Comment (muktupavels) [inline src/ui/frames.c]: Can you explain? Looking at `META_FRAME_CONTROL_MAXIMIZE` I would say that `op = META_GRAB_OP_CLICKING_UNMAXIMIZE;` is the line that should be removed... Might be good idea to check history of this code.

File: src/ui/frames.c [hunk 5]
```diff
@@ -1525,7 +1523,7 @@ meta_frames_button_press_event (GtkWidget      *widget,
           break;
         default:
           g_assert_not_reached ();
-          break;
```
Comment (muktupavels) [inline src/ui/frames.c]: Why are you removing `break`?

File: src/core/edge-resistance.c (hunk 2 above)
Comment (rbuj) [reply src/core/edge-resistance.c]: there is a break below

File: src/ui/frames.c (hunk 4 above)
Comment (rbuj) [reply src/ui/frames.c]: op is overwritten at the end of the case

File: src/ui/frames.c (hunk 5 above)
Comment (rbuj) [reply src/ui/frames.c]: default don't require break

File: src/core/edge-resistance.c (hunk 2 above)
Comment (muktupavels) [reply src/core/edge-resistance.c]: Ah, I was looking at wrong line... Sorry!

File: src/ui/frames.c (hunk 4 above)
Comment (muktupavels) [reply src/ui/frames.c]: Yes, but are you sure that is right thing? Perhaps you just found source for some bug...

File: src/core/window.c (hunk 3 above)
Comment (rbuj) [reply src/core/window.c]: great idea!

File: src/ui/frames.c (hunk 5 above)
Comment (muktupavels) [reply src/ui/frames.c]: It does not, but it is not reason to remove it, is it? I am with those who think you should always add `break` - https://softwareengineering.stackexchange.com/questions/201777/break-on-default-case-in-switch

File: src/core/window.c (hunk 3 above)
Comment (rbuj) [reply src/core/window.c]: done

**Comment by muktupavels** [comment]
Since title implies that this improves performance where are numbers before and after?

Mose if not all changes might have been already optimized by compiler... I would prefer more readable / safer code. For example initializing variable before switch. What if someone in future add new case that does not initialize variable? Sure that would be programmers problem/bug/error, but still...

File: src/core/edge-resistance.c (hunk 2 above)
Comment (rbuj) [reply src/core/edge-resistance.c]: don't worry

File: src/core/boxes.c (hunk 1 above)
Comment (rbuj) [reply src/core/boxes.c]: source code can not depend on binary optimization done by the compiler for unfound sentences like this.

Comment (muktupavels) [reply src/core/boxes.c]: I am not saying that you should relay that compiler will optimize it out. Of course this is obviously unneeded assignment and can be removed.

You are claiming to improve performance, where are numbers? And under what test conditions?

Any sane compiler will optimize out this redundant assignment so unless you disable optimizations this improves performance by exactly 0%. `-01` with gcc will be enough to optimize out redundant assignments or completely remove unused variables. So in any real life usage there wont be any performance improvement.

And even if optimizations are disabled what are you gaining? Save 1 cpu cycle (of course with modern cpus that is more complicated)?

**Pull request merged**